library/network/src/lib/y2firewall/firewalld.rb

```diff
@@ -68,18 +73,42 @@ def read
       @zones = ZoneParser.new(api.zones, api.list_all_zones).parse
       @log_denied_packets = api.log_denied_packets
       @default_zone       = api.default_zone
+      # The list of services is not read or initialized because takes time and
+      # affects to the performance and also the services are rarely touched.
       true
     end
 
     # Return from the zones list the one which matches the given name
     #
     # @param name [String] the zone name
-    # @return [Y2Firewall::Firewalld::Zone, nil] the firewalld zone with the given
-    # name
+    # @return [Y2Firewall::Firewalld::Zone, nil] the firewalld zone with the
+    # given name
     def find_zone(name)
       zones.find { |z| z.name == name }
     end
 
+    # Return from the services list the one which matches the given name
+    #
+    # @param name [String] the service name
+    # @return [Y2Firewall::Firewalld::Service, nil] the firewalld service with
+    # the given name
+    def find_service(name)
+      services.find { |s| s.name == name } || read_service(name)
+    end
+
+    # It reads the configuration of the given service or create it from scratch
+    # if not exist. After read adds it to the list of touched services.
+    #
+    # @param name [String] the service name
+    # @return [Y2Firewall::Firewalld::Service] the recently added service
+    def read_service(name)
+      service = Y2Firewall::Firewalld::Service.new(name: name)
+      service.create! if !service.supported?
```

Comment:
this looks a bit mysterious ... create a service when not supported?

Reply:
I was not sure about creating it or not. Basically not supported means the service is not defined, which implies you have to create it or deploy via a RPM.

So probably is better to raise an exception and report the error as it was in the past.